### PR TITLE
Refactoring/#1003 bump mongoose dependency

### LINKED ISSUE #1003: Update Mongoose dependency
# User Story

### As a: 
Developer

### I want:
to use the latest version of mongoose

### so that:
i'm up to date

## Acceptance criteria:
- [ ] Bump Mongoose 
- [ ] Bump Mongoose Types 
- [ ] Fix all bugs and deprecation warnings 😵

## Additional info:
dependabot has already 2 PRs on this (#991 #985), but you should update the types and the mongoose framework simultaneously. 

##Estimate:
15 h

### PR DESCRIPTION
**Pull request by dboschm**
## Description:
Bump Mongoose to new Version 5.3.14

Closes #1003
## Improvements
- [X] remove ensureIndexes deprecation Warning
- [x] fix Import ObjectId String bug
- [x] fix should duplicate courses
- [x] fix should get a directory test
- [x] fix should get a populated directory test
- [x] fix should create a sub directory test
- [x] fix should rename a directory test
- [x] Frontend Smoke Tests

### CODE REVIEW

**Comment by danielkesselberg** [comment]
image

Fix https://github.com/geli-lms/geli/pull/1035/commits/8a749ba39c7f43c31072ce5253b0693ec30530a7

File: api/src/models/User.ts [hunk 1]
```diff
@@ -209,11 +209,11 @@ userSchema.pre('remove', async function () {
   try {
     const promises = [];
     // notifications
-    promises.push(Notification.remove({user: localUser._id}));
+    promises.push(Notification.deleteMany({user: localUser._id}));
```
Comment (danielkesselberg) [inline api/src/models/User.ts]: @torss I should use `extractSingleMongoId` here?

Comment (torss) [reply api/src/models/User.ts]: @danielkesselberg I don't think that's necessary here; couldn't we even just write `{user: localUser}` and it should still be correct?

Comment (danielkesselberg) [reply api/src/models/User.ts]: I would expect it but dont know for sure :see_no_evil:

**Comment by danielkesselberg** [comment]
Good work @dboschm @torss :+1:

**Comment by dboschm** [comment]
I think we should do some frontend smoke tests before merging, to verify that there are no big new bugs.

**Pull request merged**